api/pipeline.go

```diff
@@ -0,0 +1,647 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-vela/compiler/compiler"
+
+	"github.com/go-vela/server/database"
+	"github.com/go-vela/server/router/middleware/repo"
+	"github.com/go-vela/server/source"
+	"github.com/go-vela/server/util"
+
+	"github.com/go-vela/types"
+	"github.com/go-vela/types/yaml"
+
+	"github.com/gin-gonic/gin"
+)
+
+const (
+	outputJSON = "json"
+	outputYAML = "yaml"
+)
+
+// swagger:operation GET /api/v1/pipelines/{org}/{repo} pipelines GetPipeline
+//
+// Get a pipeline configuration from the source provider
+//
+// ---
+// x-success_http_code: '200'
+// produces:
+// - application/json
+// parameters:
+// - in: path
+//   name: repo
+//   description: Name of the repo
+//   required: true
+//   type: string
+// - in: path
+//   name: org
+//   description: Name of the org
+//   required: true
+//   type: string
+// security:
+//   - ApiKeyAuth: []
+// responses:
+//   '200':
+//     description: Successfully retrieved the pipeline
+//     type: json
+//     schema:
+//       "$ref": "#/definitions/Pipeline"
+//   '400':
+//     description: Unable to retrieve the pipeline configuration templates
+//     schema:
+//       type: string
+//   '404':
+//     description: Unable to retrieve the pipeline configuration templates
+//     schema:
+//       type: string
+//   '500':
+//     description: Unable to retrieve the pipeline configuration templates
+//     schema:
+//       type: string
+
+// GetPipeline represents the API handler to capture a
+// pipeline configuration for a repo from the the source provider.
+func GetPipeline(c *gin.Context) {
+	// capture middleware values
+	m := c.MustGet("metadata").(*types.Metadata)
+	r := repo.Retrieve(c)
+
+	// capture query parameters
+	output := c.DefaultQuery("output", "yaml")
```

Comment:
Reference the `const` we created above 👍 

```suggestion
	output := c.DefaultQuery("output", outputYAML)
```